src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,257 @@
+// Import React into this file
+import React from 'react';
+// Import axios (a library for making ajax requests from our database)
+import axios from 'axios';
+
+// Import prop-types, which will validate the data of props passed between different components
+import PropTypes from 'prop-types';
+
+// Import all the Bootstrap components we will be using
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Card from 'react-bootstrap/Card';
+import Modal from 'react-bootstrap/Modal';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+// Import the MovieCard component so that we can link to it in the favourites list
+import { MovieCard } from '../movie-card/movie-card';
+
+// Import the scss file for this view
+import './profile-view.scss';
+
+// Expose the ProfileView component for use in the rest of the app
+export class ProfileView extends React.Component {
+    constructor(props) {
+        super(props);
+        // Set the state variables which will be used for updating user details
+        this.state = {
+            userDetails: [],
+            validated: false,
+            username: '',
+            password: '',
+            email: '',
+            birthDate: '',
+            favouriteMovies: [],
+            modalState: false
+        }
+
+        // Bind all the functions that are defined below to 'this'
+        this.handleFieldChange = this.handleFieldChange.bind(this);
+        this.updateUserDetails = this.updateUserDetails.bind(this);
+        this.showModal = this.showModal.bind(this);
+        this.closeModal = this.closeModal.bind(this);
+        this.deleteUserDetails = this.deleteUserDetails.bind(this);
+    }
```

Comment:
It is better to use arrow functions instead, then you will not need to bind `this`.
In your render function you can just do `onClick =() => this.myFunction(args)`